### PR TITLE
AModule: Cursor config option

### PR DESCRIPTION
**Pull request by khaneliman**
Seen a lot of discussion activity around this again... resurrecting the PR https://github.com/Alexays/Waybar/pull/3274 that resolves https://github.com/Alexays/Waybar/issues/3239. Follow up to work done in https://github.com/Alexays/Waybar/pull/3108 to make this opt in instead of always forced on users. 

One thing that might be nice is to set a global option for this... but I didn't see a way of doing that easily, atm.

### CODE REVIEW

**Comment by Alexays** [comment]
To avoid yet another breaking change, I prefer to keep the cursor change logic but make it deactivatable in the settings. WDYT?

**Comment by khaneliman** [comment]
> To avoid yet another breaking change, I prefer to keep the cursor change logic but make it deactivatable in the settings. WDYT?

Yeah, i actually agree. I was just waiting for some feedback before I changed it from the original change. I can tweak the change when I get home.

**Comment by khaneliman** [comment]
@Alexays Alright, implemented fallback to default behavior when someone doesn't provide a configuration.

File: src/AModule.cpp [hunk 1]
```diff
@@ -64,6 +68,20 @@ AModule::AModule(const Json::Value& config, const std::string& name, const std::
...
+    if (config_["cursor"].isBool() && config_["cursor"].asBool()) {
+      setCursor(Gdk::HAND2);
+    } else if (config_["cursor"].isInt()) {
+      setCursor(Gdk::CursorType(config_["cursor"].asInt()));
+    } else {
+      spdlog::warn("unknown cursor option configured on module {}", name_);
+    }
+  } else {
+    // Default behavior indicating event availability
+    setCursor(Gdk::HAND2);
```
Comment (Alexays) [inline src/AModule.cpp:83]: Should only be set if hasUserEvents_ is true

**Comment by khaneliman** [comment]
@Alexays Sorry, fixed. Also noticed the mouse enter and leave were a breaking change with tool-tips so moved it back there for the default behavior when not configured.

**Comment by khaneliman** [comment]
@Alexays Whenever you get a chance

**Comment by Alexays** [comment]
Thanks!

**Comment by haug1** [comment]
:clap:

**Comment by haug1** [comment]
I was surprised when my waybar crashed coming back from vacation. Turns out it was my own fault in the best kind of way. Had a testing configuration related to these changes left over in my config file from this:

```
  "cursor": 1231
```

As I wrote in the man page, attempting to use unavailable cursor options WILL crash your app! :joy:

**Pull request merged**